qa-tester/assets/src/js/admin-bar/components/container/index.js

```diff
@@ -0,0 +1,241 @@
+				label: __( 'Latest release', 'amp-qa-tester' ),
+				value: 'release',
+				origin: 'release',
+			},
+			{
+				label: __( 'Develop branch', 'amp-qa-tester' ),
+				value: 'develop',
+				origin: 'branch',
+			},
+		];
+
+		this.state = {
+			isLoading: true,
+			isSwitching: false,
+			isDevBuild: false,
+			error: null,
+			buildOption: { label: '', value: '' },
+			buildOptions,
+		};
+
+		this.addReleaseBranchOptions();
+		this.addPullRequestOptions();
+	}
+
+	componentWillUnmount() {
+		clearTimeout( this.errorTimeout );
+		clearTimeout( this.pageReloadTimeout );
+	}
+
+	/**
+	 * Append to the default build options a list of PRs that have build zips available for download.
+	 */
+	addReleaseBranchOptions() {
+		// Retrieve the PRs from GitHub and append them to the list of builds above.
+		getProtectedBranches().then( ( branches ) => {
+			const releaseBranchNames = branches
+				.map( ( branch ) => branch.name )
+				.filter( ( branchName ) =>
+					/[0-9]+\.[0-9]+/.test( branchName )
+				);
+
+			// We only need the release branches 1.5 and after, since only they would have built zips.
+			const releasesWithBuilds = releaseBranchNames
+				.sort()
+				.slice( releaseBranchNames.indexOf( '1.5' ) );
+
+			const branchOptions = releasesWithBuilds.map( ( branch ) => {
+				return {
+					label: `${ branch } release branch`,
+					value: branch,
+					origin: 'branch',
+				};
+			} );
+
+			this.setState( {
+				buildOptions: this.state.buildOptions.concat( branchOptions ),
+			} );
+		} );
+	}
+
+	/**
+	 * Append to the default build options a list of PRs that have build zips available for download.
+	 */
+	addPullRequestOptions() {
+		// Retrieve the PRs from GitHub and append them to the list of builds above.
+		getPullRequestsWithBuilds().then( ( results ) => {
+			const prOptions = results.map( ( pr ) => {
+				return {
+					label: `PR #${ pr.number }: ${ pr.title }`,
+					value: pr.number,
+					origin: 'pr',
+				};
+			} );
+
+			this.setState( {
+				isLoading: false,
+				buildOptions: this.state.buildOptions.concat( prOptions ),
+			} );
+		} );
+	}
+
+	/**
+	 * Set the selected build option.
+	 *
+	 * @param {Object} newOption Selected build option.
+	 */
+	handleChangeBuildOption( newOption ) {
+		this.setState( { buildOption: newOption } );
+	}
+
+	/**
+	 * Set whether a dev build is needed or not.
+	 *
+	 * @param {Event<HTMLInputElement>} event Change event.
+	 */
+	handleChangeDevBuild( event ) {
+		const checked = event.target.checked;
+		this.setState( { isDevBuild: checked } );
+	}
+
+	/**
+	 * Switches the AMP plugin to the selected build option.
+	 */
+	handleActivation() {
+		const { isDevBuild, buildOption } = this.state;
+		this.setState( { isSwitching: true } );
+
+		apiFetch( {
+			path: '/amp-qa-tester/v1/switch',
+			method: 'POST',
+			data: {
+				id: buildOption.value,
+				origin: buildOption.origin,
+				isDev: isDevBuild,
+			},
+		} )
+			.then( () => {
+				// @todo Could this be handled better?
```

Comment:
This would be a bit more complicated, but could you maybe add an endpoint to the plugin REST controller that returns whether the plugin has been activated, and ping that route every second or so until the endpoint returns `true`? Then maybe bail with an error after a few seconds if it hasn't happened.

Reply:
After revising this, I realize the issue is not with the activation of the plugin, but because some of the files from the new build have not been autoloaded as yet in time after page reload, and that leads to errors of classes and functions being missing. Delaying the page reload does solve the issue, however.

I'm wondering if we should get rid of the page reload and just simply update the build version in the admin bar instead.

Reply:
Not sure if it's just my environment or the fact that I was running it in the same environment where I'm doing AMP plugin development, but in testing again just now I had this happen after trying to activate one of branches:

<img width="1908" alt="Screen Shot 2020-06-19 at 6 01 33 PM" src="https://user-images.githubusercontent.com/9020968/85184868-160bea80-b257-11ea-8ac1-9b956e11b599.png">

Reply:
I'm not able to replicate this locally, unfortunately. Does this happen frequently?

Reply:
I wasn't able to replicate it just now in a separate environment, so this is probably a false alarm. Not sure what happened. I'll keep an eye out for it when I use the plugin in the future.